drivers/display/display_ili9340_seeed_tftv2.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2018 - 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <display.h>
+#include "display_ili9340.h"
+
+/*
+ * Derived from Seeed 2.8 inch TFT Touch Shield v2.0 sample code.
+ *
+ * https://github.com/Seeed-Studio/TFT_Touch_Shield_V2
+ */
+
+void ili9340_lcd_init(struct ili9340_data *p_ili9340)
+{
+	u8_t cmd;
+	u8_t data[15];
+
+	/* Software reset */
```

Comment:
Most display drivers I've seen use a technique like [this](https://github.com/adafruit/Adafruit_ILI9341/blob/master/Adafruit_ILI9341.cpp#L148) to specify the commands and arguments, which is shorter and more readable than a series of assignments into an array.  Would you consider moving to that form?

Reply:
> Most display drivers I've seen use a technique like [this](https://github.com/adafruit/Adafruit_ILI9341/blob/master/Adafruit_ILI9341.cpp#L148) to specify the commands and arguments, which is shorter and more readable than a series of assignments into an array. Would you consider moving to that form?

Exactly like [this](https://github.com/adafruit/Adafruit_ILI9341/blob/master/Adafruit_ILI9341.cpp#L148) should not be used. It is absolutely not clear what is initialized there.

Reply:
> Exactly like [this](https://github.com/adafruit/Adafruit_ILI9341/blob/master/Adafruit_ILI9341.cpp#L148) should not be used. It is absolutely not clear what is initialized there.

Well, no, certainly the commands which have identified names like `ILI9340_CMD_VCOM_CTRL_1` should be expressed using those names.  But most of the data values are those provided by the display manufacturer as appropriate for the way they built the device.  I'm personally fine with `0x55` vs `0x66` for `ILI9340_CMD_PIXEL_FORMAT_SET` rather than constructing it from constants like `ILI9340_DATA_PIXEL_FORMAT_RGB_16_BIT`, because the two or three people who will need to know what the value means over the maintainable life of the driver can reference the datasheet, which they'll probably need anyway.

I don't think it's worth going back and converting all the magic numbers to constructed values with named bitfields, except maybe in a case where the corresponding register can be changed at runtime, or if the chip vendor provided a header with all the constants defined in it so the mapping doesn't become another source of error.

Part of my position on this is because I spent a lot of time yesterday comparing SPI bus captures from the Zephyr application with the Adafruit reference implementation, which was a lot easier than comparing with the Zephyr code because I could immediately tell where the differences were (0x55 vs 0x66 was one of them).

But it's a choice of where you draw the line.

Reply:
Sorry. I do not have a plan to migrate to a table-style sequence now. The Seeed sample code is not in table-style and I port it plainly to Zephyr.